packages/driver/cypress/integration/commands/net_stubbing_spec.ts

```diff
@@ -2184,6 +2184,61 @@ describe('network stubbing', { retries: 2 }, function () {
             cy.get('#request').click()
             cy.get('#result').should('contain', 'client')
           })
+
+          it('works with reply', () => {
+            cy.intercept({
+              method: 'POST',
+              times: 1,
+              url: '/post-only',
+            },
+            (req) => {
+              req.reply('stubbed data')
+            }).as('interceptor')
+
+            cy.visit('fixtures/request.html')
+
+            cy.get('#request').click()
+            cy.get('#result').should('contain', 'stubbed data')
+
+            cy.get('#request').click()
+            cy.get('#result').should('contain', 'client') // Fail
+          })
+
+          it('works with reply and fallthrough', () => {
+            let times = 0
+
+            cy.intercept({
+              method: 'POST',
+              times: 3,
+              url: '/post-only',
+            },
+            (req) => {
+              req.reply(`${req.body === 'foo' ? 'foo' : 'nothing'} stubbed data ${times++}`)
+            })
+
+            cy.intercept({
+              method: 'POST',
+              times: 2,
+              url: '/post-only',
+            },
+            (req) => {
+              req.body = 'foo'
+            })
+
+            cy.visit('fixtures/request.html')
+
+            cy.get('#request').click()
+            cy.get('#result').should('contain', 'foo stubbed data 0')
+
+            cy.get('#request').click()
+            cy.get('#result').should('contain', 'foo stubbed data 1')
+
+            cy.get('#request').click()
+            cy.get('#result').should('contain', 'nothing stubbed data 2')
+
+            cy.get('#request').click()
+            cy.get('#result').should('contain', 'client') // Fail
+          })
```

Comment:
@sainthkh I added this test while checking out the fix, I believe it should pass, but it is failing. This should pass though, unless I'm missing something obvious, can you check it out?

Reply:
I also met the case that the last assertion failed, but in most cases, the test failed [with the second assertion, `foo stubbed data 1`. ](https://app.circleci.com/pipelines/github/cypress-io/cypress/22577/workflows/77244acf-f120-47e8-b2b6-5ca08e4eba95/jobs/829175)

It was happening because the second intercept is counted twice with `before:request` and `response` event. 

I learned that the solution in this PR doesn't work when there are 2+ `cy.intercept`s. So, I should drop this route and trying to fix it with your suggestion below.